apps/lockfile-explorer/src/graph/lfxGraphLoader.ts

```diff
@@ -95,58 +48,81 @@ function createLockfileDependency(
 
   if (version.startsWith('link:')) {
     const relativePath: string = version.substring('link:'.length);
-    const rootRelativePath: Path | null = new Path('.').relative(
-      new Path(containingEntry.packageJsonFolderPath).concat(relativePath)
+    const rootRelativePath: string = lockfilePath.getAbsolute(
+      containingEntry.packageJsonFolderPath,
```

Comment:
At least as of pnpm@8, `link:` protocol paths are only resolved relative to the package.json section if encountered in `importers:`. If encountered in `packages:`, they are resolved relative to the pnpm workspace root.

Reply:
Good catch. I guess those semantics are unavoidable, since the package.json folder location is nondeterministic.

Reply:
Repro here: https://github.com/octogonz/lockfile-explorer-edge-cases/commit/e586bbb60c65afe853bca5aeea91ab12d37000c3

Reply:
The fix: https://github.com/octogonz/rushstack/commit/6bebad4f0aa87826bfdecff0876aa6c11c78d4d2